node/podcontroller.go

```diff
@@ -255,10 +279,31 @@ func (pc *PodController) Run(ctx context.Context, podSyncWorkers int) error {
 // Ready returns a channel which gets closed once the PodController is ready to handle scheduled pods.
 // This channel will never close if there is an error on startup.
 // The status of this channel after shutdown is indeterminate.
+//
+// Deprecated: This is really low level and difficult to work with, you should
+// probably use `WaitReady` instead.
 func (pc *PodController) Ready() <-chan struct{} {
 	return pc.ready
 }
 
+// WaitReady waits for the pod controller to be ready.
+// As a general rule of thumb, this function will return if `Run` has returned
+// OR the controller is ready.
+//
+// This should return the same error that `Run()` would return, however it may
+// differ especially if the context (either in Run or passed in here) is
+// cancelled.
+func (pc *PodController) WaitReady(ctx context.Context) error {
```

Comment:
Should there be a way to retrieve the error after shutdown? Or wait for the controller to shutdown?

Reply:
As discussed, I added an `Err()` and `Done()` function.

Reply:
Do we need this function even with the other new functions?

Reply:
I don't think so, and FWIW, I think you need to take `pc.mu` during this function.